libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -123,15 +141,17 @@ IntentFilter obtainEventReceiverIntentFilter() {
     return eventReceiverIntentFilter;
   }
 
-  private void initializeTelemetryClient(String accessToken, String userAgent) {
+  private void initializeTelemetryClient() {
     if (isTelemetryClientInitialized()) {
       telemetryClient = createTelemetryClient(accessToken, userAgent);
       queue.setTelemetryInitialized(true);
+    } else {
+      throw new TelemetryException(ACCESS_TOKEN_USER_AGENT_EXCEPTION);
     }
   }
 
   private boolean isTelemetryClientInitialized() {
```

Comment:
Right now `isTelemetryClientInitialized` (method name) is not representing well its purpose.
We should revisit the naming.